driver-core/src/main/java/com/datastax/driver/core/Statement.java

```diff
@@ -445,4 +451,32 @@ boolean isIdempotentWithDefault(QueryOptions queryOptions) {
         else
             return queryOptions.getDefaultIdempotence();
     }
+
+    Map<String,ByteBuffer> getOutgoingPayload() {
+        return outgoingPayload;
+    }
+
+    /**
+     * Set the given outgoing payload on this statement.
+     * Each time this statement is executed, this payload will be included in the query request.
+     * <p>
+     * This method makes a defensive copy of the given map, but its values
+     * remain inherently mutable. Care should be taken not to modify the original map
+     * once it is passed to this method.
```

Comment:
The defensive copy protects us from modifications to the map itself, but users are still able to obtain a specific ByteBuffer instance from the map and modify it.